conda-store-server/tests/test_db_api.py

```diff
@@ -39,35 +39,104 @@ def test_namespace_crud(db):
 
 def test_namespace_role_mapping(db):
     namespace_name = "pytest-namespace"
+    other_namespace_name1 = "pytest-other-namespace1"
+    other_namespace_name2 = "pytest-other-namespace2"
+    other_namespace_name3 = "pytest-other-namespace3"
 
-    # starts with no namespaces for test
+    # Starts with no namespaces
     assert len(api.list_namespaces(db).all()) == 0
 
-    # create namespace
+    # Creates namespaces
     namespace = api.create_namespace(db, name=namespace_name)
+    other_namespace1 = api.create_namespace(db, name=other_namespace_name1)
+    other_namespace2 = api.create_namespace(db, name=other_namespace_name2)
+    other_namespace3 = api.create_namespace(db, name=other_namespace_name3)
     db.commit()
 
-    # check that only one namespace exists
-    assert len(api.list_namespaces(db).all()) == 1
+    # Checks that all namespaces exist
+    assert len(api.list_namespaces(db).all()) == 4
+
+    # Creates role mappings
+    api.create_namespace_role(db, name=namespace_name, other=other_namespace_name1, role="admin")
+    api.create_namespace_role(db, name=namespace_name, other=other_namespace_name2, role="admin")
+    api.create_namespace_role(db, name=namespace_name, other=other_namespace_name3, role="viewer")
+    db.commit()
 
-    # create a Role Mapping, with a failing entity
-    with pytest.raises(Exception):
-        NamespaceRoleMapping(
-            namespace=namespace, namespace_id=namespace.id, entity="invalid_entity_name"
-        )
+    # Attempts to create a role mapping with an invalid role
+    with pytest.raises(ValueError, match=r"invalid role=invalid-role"):
+        api.create_namespace_role(db, name=namespace_name, other=other_namespace_name3, role="invalid-role")
+        db.commit()
+
+    # Attempts to create a role mapping violating the uniqueness constraint
+    with pytest.raises(
+            Exception,
+            match=(r"UNIQUE constraint failed: "
+                   r"namespace_role_mapping.namespace_id, "
+                   r"namespace_role_mapping.other_namespace_id")):
+        # Runs in a nested transaction since a constraint violation will cause a rollback
+        with db.begin_nested():
+            api.create_namespace_role(db, name=namespace_name, other=other_namespace_name2, role="developer")
+            db.commit()
```

Comment:
Here's why admin above is important. I attempt to crate a new mapping but with developer. role should not be considered since we want to have one mapping for a pair of ids. So this works correctly.